src/wp-content/themes/twentytwentyone/assets/js/interactivity.js

```diff
@@ -0,0 +1,206 @@
+            setTimeout( () => {
+                var anchor = document.getElementById( event.target.hash.slice( 1 ) );
+                if ( anchor ) {
+                    anchor.scrollIntoView();
+                }
+            }, 550 );
+        },
+    },
+    callbacks: {
+        determineFocusableElements: () => {
+            if ( ! state.isPrimaryMenuOpen ) {
+                return;
+            }
+
+            const ctx = getContext();
+            const { ref } = getElement();
+			const elements = ref.querySelectorAll( 'input, a, button' );
+
+            ctx.firstFocusable = elements[ 0 ];
+            ctx.lastFocusable = elements[ elements.length - 1 ];
+        },
+
+        refreshSubmenus: () => {
+            const ctx = getContext();
+            const { ref } = getElement();
+			const elements = ref.querySelectorAll( '.sub-menu-toggle' );
+            elements.forEach( ( subMenuToggle ) => {
+                if ( ctx.activeSubmenu === subMenuToggle ) {
+                    subMenuToggle.setAttribute( 'aria-expanded', 'true' );
+                } else {
+                    subMenuToggle.setAttribute( 'aria-expanded', 'false' );
+                }
+            } );
+        },
+
+        makeIframesResponsive: () => {
+            const { ref } = getElement();
+
+            ref.querySelectorAll( 'iframe' ).forEach( function( iframe ) {
+                // Only continue if the iframe has a width & height defined.
+                if ( iframe.width && iframe.height ) {
+                    // Calculate the proportion/ratio based on the width & height.
+                    proportion = parseFloat( iframe.width ) / parseFloat( iframe.height );
+                    // Get the parent element's width.
+                    parentWidth = parseFloat( window.getComputedStyle( iframe.parentElement, null ).width.replace( 'px', '' ) );
+                    // Set the max-width & height.
+                    iframe.style.maxWidth = '100%';
+                    iframe.style.maxHeight = Math.round( parentWidth / proportion ).toString() + 'px';
+                }
+            } );
+        },
+
+        updateWindowWidthOnResize: () => {
+            // The following may be needed here since we can't use `data-wp-on--resize`?
+            const refreshWidth = () => {
+                state.windowWidth = window.innerWidth;
+            }
+            window.onresize = refreshWidth;
+        },
+
+        initDarkMode: () => {
+            let isDarkMode = window.matchMedia( '(prefers-color-scheme: dark)' ).matches;
+
+            if ( 'yes' === window.localStorage.getItem( 'twentytwentyoneDarkMode' ) ) {
+                isDarkMode = true;
+            } else if ( 'no' === window.localStorage.getItem( 'twentytwentyoneDarkMode' ) ) {
+                isDarkMode = false;
+            }
+
+            state.isDarkMode = isDarkMode;
+
+            // The following may be needed here since we can't use `data-wp-on--scroll`?
```

Comment:
`data-wp-on--scroll` works fine, but it's scoped to the element that has the directive:

```html
<div data-wp-on--scroll="callbacks.onScroll">...</div>
```

```js
// element is the <div>
element.addEventListener("scroll", callbacks.onScroll);
```

But here, you want to use the global `scroll` event: `window.addEventListener( 'scroll', checkScroll )`. 

We don't have any special syntax for directives attached to global events yet, so for now it's fine to use `data-wp-init`. Once we see how common is this, maybe we can come up with something more specific, like `data-wp-global-event--scroll` or something like that.

What do you think? 🙂

Reply:
I think having a more declarative way via attribute to add listeners for global / `window` events would be great. Maybe `data-wp-on-global--{event}` or `data-wp-on-window--{event}`? This could then work for any `window` events, like scroll, resize etc.

Reply:
Makes sense. Would you mind opening a [new discussion](https://github.com/WordPress/gutenberg/discussions/categories/interactivity-api) to talk about it? 🙂

Reply:
I've opened a related discussion: 

- https://github.com/WordPress/gutenberg/discussions/57815

Reply:
@luisherranz I've updated this PR to use the new `data-wp-on-window` and `data-wp-on-document` in https://github.com/WordPress/wordpress-develop/pull/5795/commits/44e49773bd6f4738435a91b1516027b5589b0473, reducing the amount of custom JS further. 🎉 

cc @c4rl0sbr4v0